SIGILL in test suite

**Description of the error**
A clear and concise description of what the problem is, including steps to reproduce it and the environment you are running FitBenchmarking in.

Recent PRs have been failing the tests due to SIGILL being thrown on the github runners
this causes the error code 454 and 317 and exit code 132.

**Suggested fix**
Initial investigations seem to imply that SIGILL is generally not thrown by python code 

**Additional details**
Add any other context or screenshots about the feature request here.

the following implies that the ralfit may be related to the error, but in some cases it is being thrown before ralfit is reached

#### Log showing ralfit link in unit tests
```
2026-08-19T09:23:26.1882950Z ============================= test session starts ==============================
2026-08-19T09:23:26.1885994Z platform linux -- Python 3.12.3, pytest-9.1.1, pluggy-1.6.0
2026-08-19T09:23:26.1886883Z rootdir: /__w/fitbenchmarking/fitbenchmarking
2026-08-19T09:23:26.1887701Z configfile: pyproject.toml
2026-08-19T09:23:26.1888221Z plugins: cov-7.1.0, dash-4.4.1
2026-08-19T09:23:26.1888722Z collected 946 items
2026-08-19T09:23:26.1888958Z 
2026-08-19T09:23:28.5498778Z fitbenchmarking/cli/tests/test_checkpoint_handler.py ................    [  1%]
2026-08-19T09:23:28.5557075Z fitbenchmarking/cli/tests/test_exception_handler.py ...                  [  2%]
2026-08-19T09:23:30.1506150Z fitbenchmarking/cli/tests/test_main.py ................................. [  5%]
2026-08-19T09:23:30.1934949Z .........................                                                [  8%]
2026-08-19T09:23:30.2812728Z fitbenchmarking/controllers/tests/test_controllers.py .................. [ 10%]
2026-08-19T09:23:39.5971009Z Fatal Python error: Illegal instruction
2026-08-19T09:23:39.5971440Z 
2026-08-19T09:23:39.5971658Z Thread 0x00007f2f238606c0 (most recent call first):
2026-08-19T09:23:39.5972282Z   File "/usr/lib/python3.12/threading.py", line 359 in wait
2026-08-19T09:23:39.5973361Z   File "/usr/lib/python3.12/threading.py", line 655 in wait
2026-08-19T09:23:39.5974405Z   File "/opt/venv/lib/python3.12/site-packages/tqdm/_monitor.py", line 69 in run
2026-08-19T09:23:39.5975279Z   File "/usr/lib/python3.12/threading.py", line 1073 in _bootstrap_inner
2026-08-19T09:23:39.5976072Z   File "/usr/lib/python3.12/threading.py", line 1030 in _bootstrap
2026-08-19T09:23:39.5976530Z 
2026-08-19T09:23:39.5976777Z Current thread 0x00007f2f49f58140 (most recent call first):
2026-08-19T09:23:39.5977811Z   File "/__w/fitbenchmarking/fitbenchmarking/fitbenchmarking/controllers/ralfit_controller.py", line 177 in fit
2026-08-19T09:23:39.5979344Z   File "/__w/fitbenchmarking/fitbenchmarking/fitbenchmarking/controllers/tests/test_controllers.py", line 994 in test_controller_bounds
2026-08-19T09:23:39.5980854Z   File "/opt/venv/lib/python3.12/site-packages/parameterized/parameterized.py", line 620 in standalone_func
2026-08-19T09:23:39.5981942Z   File "/usr/lib/python3.12/unittest/case.py", line 589 in _callTestMethod
2026-08-19T09:23:39.5982689Z   File "/usr/lib/python3.12/unittest/case.py", line 634 in run
2026-08-19T09:23:39.5983377Z   File "/usr/lib/python3.12/unittest/case.py", line 690 in __call__
2026-08-19T09:23:39.5984381Z   File "/opt/venv/lib/python3.12/site-packages/_pytest/unittest.py", line 410 in runtest
2026-08-19T09:23:39.5985440Z   File "/opt/venv/lib/python3.12/site-packages/_pytest/runner.py", line 184 in pytest_runtest_call
2026-08-19T09:23:39.5986526Z   File "/opt/venv/lib/python3.12/site-packages/pluggy/_callers.py", line 121 in _multicall
2026-08-19T09:23:39.5987497Z   File "/opt/venv/lib/python3.12/site-packages/pluggy/_manager.py", line 120 in _hookexec
2026-08-19T09:23:39.5988444Z   File "/opt/venv/lib/python3.12/site-packages/pluggy/_hooks.py", line 512 in __call__
2026-08-19T09:23:39.5989354Z   File "/opt/venv/lib/python3.12/site-packages/_pytest/runner.py", line 250 in <lambda>
2026-08-19T09:23:39.5990316Z   File "/opt/venv/lib/python3.12/site-packages/_pytest/runner.py", line 361 in from_call
2026-08-19T09:23:39.5991347Z   File "/opt/venv/lib/python3.12/site-packages/_pytest/runner.py", line 249 in call_and_report
2026-08-19T09:23:39.5992417Z   File "/opt/venv/lib/python3.12/site-packages/_pytest/runner.py", line 139 in runtestprotocol
2026-08-19T09:23:39.5995502Z   File "/opt/venv/lib/python3.12/site-packages/_pytest/runner.py", line 118 in pytest_runtest_protocol
2026-08-19T09:23:39.5996422Z   File "/opt/venv/lib/python3.12/site-packages/pluggy/_callers.py", line 121 in _multicall
2026-08-19T09:23:39.5997278Z   File "/opt/venv/lib/python3.12/site-packages/pluggy/_manager.py", line 120 in _hookexec
2026-08-19T09:23:39.5997883Z   File "/opt/venv/lib/python3.12/site-packages/pluggy/_hooks.py", line 512 in __call__
2026-08-19T09:23:39.5998478Z   File "/opt/venv/lib/python3.12/site-packages/_pytest/main.py", line 408 in pytest_runtestloop
2026-08-19T09:23:39.5999087Z   File "/opt/venv/lib/python3.12/site-packages/pluggy/_callers.py", line 121 in _multicall
2026-08-19T09:23:39.5999649Z   File "/opt/venv/lib/python3.12/site-packages/pluggy/_manager.py", line 120 in _hookexec
2026-08-19T09:23:39.6000190Z   File "/opt/venv/lib/python3.12/site-packages/pluggy/_hooks.py", line 512 in __call__
2026-08-19T09:23:39.6000711Z   File "/opt/venv/lib/python3.12/site-packages/_pytest/main.py", line 384 in _main
2026-08-19T09:23:39.6001259Z   File "/opt/venv/lib/python3.12/site-packages/_pytest/main.py", line 330 in wrap_session
2026-08-19T09:23:39.6001862Z   File "/opt/venv/lib/python3.12/site-packages/_pytest/main.py", line 377 in pytest_cmdline_main
2026-08-19T09:23:39.6002449Z   File "/opt/venv/lib/python3.12/site-packages/pluggy/_callers.py", line 121 in _multicall
2026-08-19T09:23:39.6003014Z   File "/opt/venv/lib/python3.12/site-packages/pluggy/_manager.py", line 120 in _hookexec
2026-08-19T09:23:39.6003554Z   File "/opt/venv/lib/python3.12/site-packages/pluggy/_hooks.py", line 512 in __call__
2026-08-19T09:23:39.6004109Z   File "/opt/venv/lib/python3.12/site-packages/_pytest/config/__init__.py", line 229 in _main
2026-08-19T09:23:39.6004861Z   File "/opt/venv/lib/python3.12/site-packages/_pytest/config/__init__.py", line 253 in _console_main
2026-08-19T09:23:39.6005342Z   File "/opt/venv/bin/pytest", line 8 in <module>
2026-08-19T09:23:39.6005563Z 
2026-08-19T09:23:39.6023509Z Extension modules: markupsafe._speedups, charset_normalizer.md, charset_normalizer.cd, requests.packages.charset_normalizer.md, requests.packages.chardet.md, requests.packages.charset_normalizer.cd, requests.packages.chardet.cd, psutil._psutil_linux, numpy._core._multiarray_umath, numpy.linalg._umath_linalg, numpy.random._common, numpy.random.bit_generator, numpy.random._bounded_integers, numpy.random._pcg64, numpy.random._generator, numpy.random._mt19937, numpy.random._philox, numpy.random._sfc64, numpy.random.mtrand, pandas._libs.tslibs.ccalendar, pandas._libs.tslibs.np_datetime, pandas._libs.tslibs.dtypes, pandas._libs.tslibs.base, pandas._libs.tslibs.nattype, pandas._libs.tslibs.timezones, pandas._libs.tslibs.fields, pandas._libs.tslibs.timedeltas, pandas._libs.tslibs.tzconversion, pandas._libs.tslibs.timestamps, pandas._libs.properties, pandas._libs.tslibs.offsets, pandas._libs.tslibs.strptime, pandas._libs.tslibs.parsing, pandas._libs.tslibs.conversion, pandas._libs.tslibs.period, pandas._libs.tslibs.vectorized, pandas._libs.ops_dispatch, pandas._libs.missing, pandas._libs.hashtable, pandas._libs.algos, pandas._libs.interval, pandas._libs.lib, pandas._libs.ops, pandas._libs.hashing, pandas._libs.arrays, pandas._libs.tslib, pandas._libs.sparse, pandas._libs.internals, pandas._libs.indexing, pandas._libs.index, pandas._libs.writers, pandas._libs.join, pandas._libs.window.aggregations, pandas._libs.window.indexers, pandas._libs.reshape, pandas._libs.groupby, pandas._libs.json, pandas._libs.parsers, pandas._libs.testing, PIL._imaging, cython.cimports.libc.math, scipy._lib._ccallback_c, scipy.sparse._sparsetools, _csparsetools, scipy.sparse._csparsetools, scipy.linalg._fblas, scipy.linalg._flapack, scipy.linalg.cython_lapack, scipy.linalg._cythonized_array_utils, scipy.linalg._solve_toeplitz, scipy.linalg._decomp_lu_cython, scipy.linalg._matfuncs_sqrtm_triu, scipy.linalg._matfuncs_expm, scipy.linalg._linalg_pythran, scipy.linalg.cython_blas, scipy.linalg._decomp_update, scipy.sparse.linalg._dsolve._superlu, scipy.sparse.linalg._eigen.arpack._arpack, scipy.sparse.linalg._propack._spropack, scipy.sparse.linalg._propack._dpropack, scipy.sparse.linalg._propack._cpropack, scipy.sparse.linalg._propack._zpropack, scipy.sparse.csgraph._tools, scipy.sparse.csgraph._shortest_path, scipy.sparse.csgraph._traversal, scipy.sparse.csgraph._min_spanning_tree, scipy.sparse.csgraph._flow, scipy.sparse.csgraph._matching, scipy.sparse.csgraph._reordering, scipy.spatial._ckdtree, scipy._lib.messagestream, scipy.spatial._qhull, scipy.spatial._voronoi, scipy.spatial._distance_wrap, scipy.spatial._hausdorff, scipy.special._ufuncs_cxx, scipy.special._ufuncs, scipy.special._specfun, scipy.special._comb, scipy.special._ellip_harm_2, scipy.spatial.transform._rotation, scipy.optimize._group_columns, scipy.optimize._trlib._trlib, sksparse.cholmod, scipy.optimize._lbfgsb, _moduleTNC, scipy.optimize._moduleTNC, scipy.optimize._cobyla, scipy.optimize._slsqp, scipy.optimize._minpack, scipy.optimize._lsq.givens_elimination, scipy.optimize._zeros, scipy.optimize._cython_nnls, scipy._lib._uarray._uarray, scipy.linalg._decomp_interpolative, scipy.optimize._bglu_dense, scipy.optimize._lsap, scipy.optimize._direct, scipy.integrate._odepack, scipy.integrate._quadpack, scipy.integrate._vode, scipy.integrate._dop, scipy.integrate._lsoda, scipy.interpolate._fitpack, scipy.interpolate._dfitpack, scipy.interpolate._dierckx, scipy.interpolate._ppoly, scipy.interpolate._interpnd, scipy.interpolate._rbfinterp_pythran, scipy.interpolate._rgi_cython, scipy.interpolate._bspl, scipy.special.cython_special, scipy.stats._stats, scipy.stats._sobol, scipy.stats._qmc_cy, scipy.stats._biasedurn, scipy.stats._stats_pythran, scipy.stats._levy_stable.levyst, scipy.stats._ansari_swilk_statistics, scipy.stats._mvn, scipy.stats._rcont.rcont, scipy.ndimage._nd_image, scipy.ndimage._rank_filter_1d, _ni_label, scipy.ndimage._ni_label, kiwisolver._cext, lxml._elementpath, lxml.etree, rapidfuzz._feature_detector_cpp, rapidfuzz.distance._initialize_cpp, rapidfuzz.distance.metrics_cpp_avx2, rapidfuzz.fuzz_cpp_avx2, rapidfuzz.process_cpp_impl, rapidfuzz.utils_cpp, nlopt._nlopt, PIL._imagingmath, yaml._yaml, numba.core.typeconv._typeconv, numba._helperlib, numba._dynfunc, numba._dispatcher, numba.core.runtime._nrt_python, numba.np.ufunc._internal, numba.experimental.jitclass._box, galahad.arc, galahad.bgo, galahad.dgo, galahad.nls, galahad.trb, galahad.tru, h5py._errors, h5py.defs, h5py._objects, h5py.h5, h5py.utils, h5py.h5t, h5py.h5s, h5py.h5ac, h5py.h5p, h5py.h5r, h5py._npystrings, h5py._proxy, h5py._conv, h5py.h5z, h5py.h5a, h5py.h5d, h5py.h5ds, h5py.h5g, h5py.h5i, h5py.h5o, h5py.h5f, h5py.h5fd, h5py.h5pl, h5py.h5l, h5py._selector, ral_nlls (total: 196)
2026-08-19T09:23:41.7705323Z ci/unit_tests.sh: line 6:   454 Illegal instruction     (core dumped) pytest $FULL_DIRS $COV_ARGS --cov-report lcov --cov-report term-missing --junit-xml test-results/full_unit_pytest.xml --durations=0
2026-08-19T09:23:41.7740268Z .........................................
2026-08-19T09:23:41.7761110Z ##[error]Process completed with exit code 1.
```
### Log showing instant fail on systests
```
============================= test session starts ==============================
platform linux -- Python 3.12.3, pytest-9.1.1, pluggy-1.6.0
rootdir: /__w/fitbenchmarking/fitbenchmarking
configfile: pyproject.toml
plugins: cov-7.1.0, dash-4.4.1
collected 7 items

ci/system_tests.sh: line 3:   317 Illegal instruction     (core dumped) pytest fitbenchmarking/systests --junit-xml test-results/full_system_pytest.xml --durations=0
fitbenchmarking/systests/test_regression.py 
Error: Process completed with exit code 132.
```

--- a/Containers/full-tests/Containerfile
+++ b/Containers/full-tests/Containerfile
@@ -95,17 +95,18 @@ RUN mkdir -p /RALFit-build
 WORKDIR /RALFit-build
 
 ENV CC=gcc
-# RALFit defaults CMAKE_BUILD_TYPE to Release, and its Release flags hardcode
-# -march=native. That targets the CPU of whichever runner builds this image, so
-# libral_nlls.so dies with SIGILL on any machine with a narrower instruction set.
-# RelWithDebInfo is used because RALFit set()s CMAKE_<LANG>_FLAGS_RELEASE as a
-# normal variable, which overrides anything passed here with -D; RelWithDebInfo
-# does not match its "Release" check, so these flags are honoured instead.
-RUN cmake /RALFit/libRALFit \
-      -DCMAKE_BUILD_TYPE=RelWithDebInfo \
-      -DCMAKE_C_FLAGS_RELWITHDEBINFO="-O2 -march=x86-64-v3 -fno-fast-math -fopenmp" \
-      -DCMAKE_Fortran_FLAGS_RELWITHDEBINFO="-O2 -march=x86-64-v3 -fno-fast-math -fopenmp" \
-    && make && make install
+# RALFit defaults CMAKE_BUILD_TYPE to Release and hardcodes -march=native in its
+# Release flags. That targets the CPU of whichever runner builds this image, so
+# libral_nlls.so dies with SIGILL on any machine with a narrower instruction set
+# (e.g. an image built on an AVX-512 runner, then run on an AVX2-only machine).
+# The flag is patched out rather than overridden on the command line: RALFit
+# set()s CMAKE_<LANG>_FLAGS_RELEASE as a normal variable, so -D values are
+# discarded, and it rejects any build type other than Debug or Release.
+# x86-64-v3 is AVX2+FMA, supported by anything from Haswell (2013) onwards.
+# The grep afterwards asserts no -march=native survived the substitution.
+RUN sed -i 's/-march=native/-march=x86-64-v3/g' /RALFit/libRALFit/CMakeLists.txt \
+    && ! grep -q -- '-march=native' /RALFit/libRALFit/CMakeLists.txt
+RUN cmake /RALFit/libRALFit && make && make install
 
 # make install for RALFit doesn't work in a container, so manually pip install
 RUN pip install --no-cache-dir .

--- a/Containers/full-tests/Containerfile
+++ b/Containers/full-tests/Containerfile
@@ -95,7 +95,17 @@ RUN mkdir -p /RALFit-build
 WORKDIR /RALFit-build
 
 ENV CC=gcc
-RUN cmake /RALFit/libRALFit && make && make install
+# RALFit defaults CMAKE_BUILD_TYPE to Release, and its Release flags hardcode
+# -march=native. That targets the CPU of whichever runner builds this image, so
+# libral_nlls.so dies with SIGILL on any machine with a narrower instruction set.
+# RelWithDebInfo is used because RALFit set()s CMAKE_<LANG>_FLAGS_RELEASE as a
+# normal variable, which overrides anything passed here with -D; RelWithDebInfo
+# does not match its "Release" check, so these flags are honoured instead.
+RUN cmake /RALFit/libRALFit \
+      -DCMAKE_BUILD_TYPE=RelWithDebInfo \
+      -DCMAKE_C_FLAGS_RELWITHDEBINFO="-O2 -march=x86-64-v3 -fno-fast-math -fopenmp" \
+      -DCMAKE_Fortran_FLAGS_RELWITHDEBINFO="-O2 -march=x86-64-v3 -fno-fast-math -fopenmp" \
+    && make && make install
 
 # make install for RALFit doesn't work in a container, so manually pip install
 RUN pip install --no-cache-dir .

